inc/rest-api.php

```diff
@@ -165,6 +165,24 @@ function rest_user_search( \WP_REST_Request $request ) {
 	return rest_ensure_response( $data );
 }
 
+/**
+ * Filter REST API dispath request to skip ES integration. Disables VIP
+ * Enterprise Search for Profiles to support Byline Manager's byline and user
+ * selectors in Gutenberg.
+ *
+ * @param mixed            $result Response to replace the requested version with.
+ * @param \WP_REST_Server  $server Server instance.
+ * @param \WP_REST_Request $request Request used to generate the response.
+ * @return mixed Filtered response.
+ */
+function filter_rest_pre_dispatch( $result, $server, $request ) {
+	if ( '/byline-manager/v1/authors' === $request->get_route() ) {
+		add_filter( 'ep_skip_query_integration', '__return_true' );
+	}
+	return $result;
+}
+add_filter( 'rest_pre_dispatch', __NAMESPACE__ . '\filter_rest_pre_dispatch', 10, 3 );
```

Comment:
I have contrary feedback about this change: as an open-source plugin, I don't think we should be making this decision for all users.

If I decide to use the VIP Enterprise Search here, this is creating a new task for me to unhook this. I could be unaware of this change, as most users will be, and assume it is using Elasticsearch when it is not.

The root of the issue stands for this line, https://github.com/alleyinteractive/byline-manager/blob/a51e6d27b38cdc3a3d2bd5a879303077317cdcf6/inc/rest-api.php#L64, together with the `	define( 'VIP_ENABLE_VIP_SEARCH_QUERY_INTEGRATION', true );` flag VIP provides, which forces all search queries to use VIP Search.

If I set this to `false`, I think it should stop this from happening. Essentially, someone is making a decision to run ES in all search queries unaware of its reach.

--

My suggestion would be to not add this to the plugin and leave the site itself to handle the ElasticSearch integration.

Reply:
@renatonascalves @joemcgill - Personally, I don't have a dog in this fight. I can see value in either approach. However, I would like to get the REST API support change merged to the `production` to complete Block Editor support. This is currently a blocker on another project. Recognizing that the broken Byline Selector that Joe noted earlier in this PR is due to ES integration, we can isolate that as a separate conversation to be continued apart from this PR.